### PR TITLE
PHOENIX-6827 Update hbase-version to 2.4.15 in phoenix master branch

### PR DESCRIPTION
**Pull request by shahrs87**

### CODE REVIEW

**Comment by virajjasani** [comment]
Looks like end2end tests are not run https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1523/1/testReport/

Perhaps we can make one liner change to phoenix-core pom and that would trigger full test suite. Once test results are available, we can remove the one liner change in phoenix-core pom.

**Comment by virajjasani** [comment]
This is weird, we already use hadoop 3 built hbase 2 artifacts so this should not have happened. @stoty @richardantal any idea what is wrong here?

File: phoenix-core/src/test/java/org/apache/phoenix/mapreduce/index/BaseIndexTest.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
  * Creates a simple data table and index table
  *
  */
-public class BaseIndexTest extends BaseConnectionlessQueryTest {
+public class  BaseIndexTest extends BaseConnectionlessQueryTest {
```
Comment (virajjasani) [inline phoenix-core/src/test/java/org/apache/phoenix/mapreduce/index/BaseIndexTest.java:39]: Btw somehow this change also didn't trigger end2end tests. Perhaps adding new line in phoenix-core/pom.xml would do.

**Comment by stoty** [comment]
The Yetus script doesn't handle updating the HBase version in a patch.
The code I added to avoid recompiling HBase multiple times doesn't check if the version has changed.

You can ignore the test error, @virajjasani .

**Comment by stoty** [comment]
I'm not sure why test weren't triggered, maybe only the non-test code is checked by Yetus ?

**Comment by virajjasani** [comment]
> I'm not sure why test weren't triggered, maybe only the non-test code is checked by Yetus ?

Exactly, I feel the same. Hence, i usually just add one line to phoenix-core pom

**Comment by shahrs87** [comment]
Added one line change to phoenix-core/pom.xml. Lets see if this triggers end2end tests.

**Review by virajjasani — APPROVED** [review]

**Comment by virajjasani** [comment]
Unfortunately, we still don't have end2end tests

**Review by chrajeshbabu — COMMENTED** [review]
Hi

**Comment by virajjasani** [comment]
@stoty shall we merge this PR and let pre-commit run builds with 2.4 profile to verify results later?

**Comment by stoty** [comment]
Yes, @virajjasani.

**Pull request merged**